modules/api/src/main/java/org/apache/fluo/api/data/Bytes.java

```diff
@@ -355,6 +354,36 @@ public static final Bytes of(String s, Charset c) {
   }
 
   /**
+   * Checks if this has the passed prefix
+   * 
+   * @param prefix
+   * @return true or false
+   */
+  public boolean startsWith(Bytes prefix) {
+    for (int i = 0; i < prefix.length; i++) {
+      if (this.byteAt(i) != prefix.byteAt(i))
+        return false;
+    }
+    return true;
+  }
+
+  /**
+   * Checks if this has the passed suffix
+   * 
+   * @param suffix
+   * @return true or false
+   */
+  public boolean endsWith(Bytes suffix) {
+    int startOffset = this.length - suffix.length;
+    for (int i = startOffset; i < this.length; i++) {
```

Comment:
if startOffset is negative here, this should return false. In other words, it is always impossible for a string to be a suffix of a shorter string.